kong/templates/kong_defaults.lua

```diff
@@ -168,6 +168,7 @@ dns_cache_size = 10000
 dns_not_found_ttl = 30
 dns_error_ttl = 1
 dns_no_sync = off
+legacy_dns_client = off
```

Comment:
```suggestion
legacy_dns_client = on
```
The plan is to have the new DNS client as default only on 4.0.

Reply:
yeah, will update it

Reply:
hi @locao , as we talked with @AndyZhang0707 , we will enable it by default in master branch. When preparing for new release, the PM will finally determine whether to keep it enabled. If we need to disable it, we will create a new new PR to do so.